## 2016 Rio Olympics

In [57]:
op = pd.read_csv('data/athletes.csv')
ct = pd.read_csv('data/countries.csv')
op.tail()

id                name nationality     sex       dob  height  \
11533  265605954  Zurian Hechavarria         CUB  female   8/10/95    1.64   
11534  214461847      Zuzana Hejnova         CZE  female  12/19/86    1.73   
11535   88361042             di Xiao         CHN    male   5/14/91    1.85   
11536  900065925   le Quoc Toan Tran         VIE    male    4/5/89    1.60   
11537  711404576      le Roux Hamman         RSA    male    1/6/92    1.85   

       weight          sport  gold  silver  bronze  
11533    58.0      athletics     0       0       0  
11534    63.0      athletics     0       0       0  
11535   100.0      wrestling     0       0       0  
11536    56.0  weightlifting     0       0       0  
11537    70.0      athletics     0       0       0

In [6]:
op.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11538 entries, 0 to 11537
Data columns (total 11 columns):
id             11538 non-null int64
name           11538 non-null object
nationality    11538 non-null object
sex            11538 non-null object
dob            11537 non-null object
height         11208 non-null float64
weight         10879 non-null float64
sport          11538 non-null object
gold           11538 non-null int64
silver         11538 non-null int64
bronze         11538 non-null int64
dtypes: float64(2), int64(4), object(5)
memory usage: 991.6+ KB


1. `id`
    * Athlete id
    * Integer between 1 and 1,000,000,000
    * Unique
    * No null values
2. `name`
    * Athlete's full name
    * String up to forty characters in length
    * Not unique
    * No null values
3. `nationality`
    * Athlete's nationality
    * One of the [IOC][ioc]'s 206 [three-letter country codes][tcc], or `ROT` for members of the [Refugee Olympic Team][rot]. Kuwaiti athletes' nationality is given as `IOA` due to the [suspension of the Kuwait Olympic Committee][koc]
    * Not unique
    * No null values
4. `sex`
    * Athlete's sex
    * One of two lower-case string values:
        * `male`
        * `female`
    * Not unique
    * No null values
5. `date_of_birth`
    * Athlete's date of birth
    * `YYYY-MM-DD` format
    * Not unique
    * No null values
6. `height`
    * Athlete's height, in metres
    * Floating-point number
    * Not unique
    * Contains null values
7. `weight`
    * Athlete's weight, in kilograms
    * Integer
    * Not unique
    * Contains null values
8. `sport`
    * The sport in which the athlete competes, as defined by the [IOC][ioc]
    * One of 28 lower-case string values
        * `aquatics`
        * `archery`
        * `athletics`
        * `badminton`
        * `basketball`
        * `boxing`
        * `canoe`
        * `cycling`
        * `equestrian`
        * `fencing`
        * `football`
        * `golf`
        * `gymnastics`
        * `handball`
        * `hockey`
        * `judo`
        * `modern pentathlon`
        * `rowing`
        * `rugby sevens`
        * `sailing`
        * `shooting`
        * `table tennis`
        * `taekwondo`
        * `tennis`
        * `triathlon`
        * `volleyball`
        * `weightlifting`
        * `wrestling`
    * Not unique
    * No null values
9. `gold`
    * Number of gold medals won by the athlete
    * Integer
    * Not unique
    * No null values
10. `silver`
    * Number of silver medals won by the athlete
    * Integer
    * Not unique
    * No null values
11. `bronze`
    * Number of bronze medals won by the athlete
    * Integer
    * Not unique
    * No null values
12. `info`
    * Free-form English-language description of the athlete
    * String
    * Unique (excluding null values)
    * Contains null values

#### Note :

1. 총 11,538개 데이터
2. 키와 몸무게에 NaN값 발생

In [7]:
op.describe()

id        height        weight          gold        silver  \
count  1.153800e+04  11208.000000  10879.000000  11538.000000  11538.000000   
mean   4.999885e+08      1.766282     72.068205      0.057722      0.056769   
std    2.908648e+08      0.112719     16.177334      0.255910      0.239147   
min    1.834700e+04      1.210000     31.000000      0.000000      0.000000   
25%    2.450997e+08      1.690000     60.000000      0.000000      0.000000   
50%    5.002011e+08      1.760000     70.000000      0.000000      0.000000   
75%    7.539874e+08      1.840000     81.000000      0.000000      0.000000   
max    9.999878e+08      2.210000    170.000000      5.000000      2.000000   

             bronze  
count  11538.000000  
mean       0.061016  
std        0.243320  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        0.000000  
max        2.000000

#### Note :

1. 키는 평균 1.76m (176cm), 최소 1.21m부터 최대 2.21m
2. 몸무게는 평균 72.06kg, 최소 31kg부터 최대 170kg까지
3. 금메달을 최대한 많이 획득한 사람은 5개

In [30]:
for num in range(0, len(op.columns)):
    print('uniqe {} : '.format(op.columns[num]), len(op[op.columns[num]].unique()))   

uniqe id :  11538
uniqe name :  11517
uniqe nationality :  207
uniqe sex :  2
uniqe dob :  5596
uniqe height :  83
uniqe weight :  124
uniqe sport :  28
uniqe gold :  6
uniqe silver :  3
uniqe bronze :  3


## 내가 해보고 싶은거 해보자

### Q1. 종목별로 어떤 특징을 가진 선수들이 메달을 획득할까?

### Q2. 종목에 상관없이 메달을 딴 선수들간의 특징이 있을까?

### Q3. 남/여 별로 차이가 있을까?

In [31]:
op.groupby(op.sport).count()

id  name  nationality   sex   dob  height  weight  gold  \
sport                                                                          
aquatics           1445  1445         1445  1445  1445    1396    1399  1445   
archery             128   128          128   128   128     126     126   128   
athletics          2363  2363         2363  2363  2363    2196    2200  2363   
badminton           172   172          172   172   172     170     164   172   
basketball          288   288          288   288   288     288     284   288   
boxing              286   286          286   286   286     275       0   286   
canoe               331   331          331   331   331     328     327   331   
cycling             525   525          525   525   525     504     502   525   
equestrian          222   222          222   222   222     216     215   222   
fencing             246   246          246   246   246     245     245   246   
football            611   611          611   611   611     601     583   611   
golf                120   120          120   120   120     117     115   120   
gymnastics          324   324          324   324   324     324     319   324   
handball            363   363          363   363   363     363     349   363   
hockey              432   432          432   432   432     432     432   432   
judo                392   392          392   392   392     382     380   392   
modern pentathlon    72    72           72    72    72      72      72    72   
rowing              547   547          547   547   547     535     535   547   
rugby sevens        300   300          300   300   300     297     297   300   
sailing             380   380          380   380   379     372     372   380   
shooting            390   390          390   390   390     383     380   390   
table tennis        172   172          172   172   172     168     169   172   
taekwondo           128   128          128   128   128     127     125   128   
tennis              196   196          196   196   196     191     191   196   
triathlon           110   110          110   110   110     109     109   110   
volleyball          384   384          384   384   384     383     382   384   
weightlifting       258   258          258   258   258     258     258   258   
wrestling           353   353          353   353   353     350     349   353   

                   silver  bronze  
sport                              
aquatics             1445    1445  
archery               128     128  
athletics            2363    2363  
badminton             172     172  
basketball            288     288  
boxing                286     286  
canoe                 331     331  
cycling               525     525  
equestrian            222     222  
fencing               246     246  
football              611     611  
golf                  120     120  
gymnastics            324     324  
handball              363     363  
hockey                432     432  
judo                  392     392  
modern pentathlon      72      72  
rowing                547     547  
rugby sevens          300     300  
sailing               380     380  
shooting              390     390  
table tennis          172     172  
taekwondo             128     128  
tennis                196     196  
triathlon             110     110  
volleyball            384     384  
weightlifting         258     258  
wrestling             353     353

#### aquatics (수중)

In [42]:
aquatics = op[op.sport == 'aquatics']
aquatics.tail()

id               name nationality     sex      dob  height  \
11500  274387882     Ziv Kalontarov         ISR    male  1/15/97    1.85   
11503  593183472        Zivko Gocic         SRB    male  8/22/82    1.93   
11507  446963221       Zoe Arancini         AUS  female  7/14/91    1.70   
11512  371106012      Zohar Shikler         ISR  female   7/8/97    1.78   
11530  739315046  Zsuzsanna Jakabos         HUN  female   4/3/89    1.85   

       weight     sport  gold  silver  bronze  
11500    71.0  aquatics     0       0       0  
11503    93.0  aquatics     1       0       0  
11507    70.0  aquatics     0       0       0  
11512    61.0  aquatics     0       0       0  
11530    70.0  aquatics     0       0       0

#### archery (양궁)

In [44]:
archery = op[op.sport == 'archery']
archery.tail()

id                name nationality     sex      dob  height  \
11347  480817405           Yuhong Qi         CHN  female  8/25/89    1.66   
11356  384346925        Yuki Hayashi         JPN  female  10/2/84    1.63   
11381  323145387  Yuliya Lobzhenidze         GEO  female  8/23/77    1.77   
11436  306067063        Zach Garrett         USA    male   4/8/95    1.78   
11440  242878558        Zahra Nemati         IRI  female  4/30/85    1.76   

       weight    sport  gold  silver  bronze  
11347    78.0  archery     0       0       0  
11356    65.0  archery     0       0       0  
11381    80.0  archery     0       0       0  
11436    63.0  archery     0       1       0  
11440    60.0  archery     0       0       0

#### athletics (육상)

In [45]:
athletics = op[op.sport == 'athletics']
athletics.tail()

id                name nationality     sex       dob  height  \
11520  724419150       Zouhair Aouad         BRN    male    4/7/89    1.75   
11525  999437858      Zsofia Erdelyi         HUN  female  12/10/87    1.64   
11533  265605954  Zurian Hechavarria         CUB  female   8/10/95    1.64   
11534  214461847      Zuzana Hejnova         CZE  female  12/19/86    1.73   
11537  711404576      le Roux Hamman         RSA    male    1/6/92    1.85   

       weight      sport  gold  silver  bronze  
11520    69.0  athletics     0       0       0  
11525    53.0  athletics     0       0       0  
11533    58.0  athletics     0       0       0  
11534    63.0  athletics     0       0       0  
11537    70.0  athletics     0       0       0

#### badminton (배드민턴)

In [46]:
badminton = op[op.sport == 'badminton']
badminton.tail()

id                 name nationality     sex      dob  height  \
11331  247982546        Yuanting Tang         CHN  female   8/2/94    1.75   
11344  928265049            Yuhan Tan         BEL    male  4/21/87    1.82   
11390  660495063               Yun Hu         HKG    male  8/31/81    1.78   
11392  491896324          Yunlei Zhao         CHN  female  8/25/86    1.73   
11492  955970530  Zi Liang Derek Wong         SIN    male  1/13/89    1.76   

       weight      sport  gold  silver  bronze  
11331    70.0  badminton     0       0       0  
11344    74.0  badminton     0       0       0  
11390    73.0  badminton     0       0       0  
11392    60.0  badminton     0       0       1  
11492    69.0  badminton     0       0       0

#### basketball (농구)

In [47]:
basketball = op[op.sport == 'basketball']
basketball.tail()

id              name nationality     sex      dob  height  \
11361  223316076     Yuki Miyazawa         JPN  female   6/2/93    1.82   
11384  521174364  Yuliya Rytsikava         BLR  female   9/8/86    1.80   
11460  916408500      Zeljko Sakic         CRO    male  4/14/88    2.03   
11475  167115740       Zhelin Wang         CHN    male  1/20/94    2.14   
11483  756808595      Zhifang Zhao         CHN  female  8/11/94    1.68   

       weight       sport  gold  silver  bronze  
11361    70.0  basketball     0       0       0  
11384    69.0  basketball     0       0       0  
11460   105.0  basketball     0       0       0  
11475   100.0  basketball     0       0       0  
11483    57.0  basketball     0       0       0

In [54]:
basketball.groupby('nationality').sum()

id  height  weight  gold  silver  bronze
nationality                                                   
ARG           8244443500   23.91  1176.0     0       0       0
AUS          13208276353   46.03  2126.0     0       0       0
BLR           5274076284   21.99   891.0     0       0       0
BRA          15437720658   45.71  2164.0     0       0       0
CAN           5418170072   22.01   888.0     0       0       0
CHN          11671142844   46.78  2135.0     0       0       0
CRO           5551807926   24.23  1224.0     0       0       0
ESP          11958143215   46.04  2017.0     0      12      12
FRA          11554918479   45.93  2035.0     0       0       0
JPN           5470759669   21.22   841.0     0       0       0
LTU           7218735455   24.23  1123.0     0       0       0
NGR           4513957637   23.96  1066.0     0       0       0
SEN           6638378906   21.85   917.0     0       0       0
SRB          12791745976   46.31  2105.0     0      12      12
TUR           4750820327   22.14   909.0     0       0       0
USA          10212393990   46.60  2180.0    24       0       0
VEN           5123621260   23.65  1124.0     0       0       0

#### boxing (권투)

In [55]:
boxing = op[op.sport == 'boxing']
boxing.tail()

id                  name nationality     sex       dob  height  \
11465  980866226   Zhaina Shekerbekova         KAZ  female  12/17/89    1.54   
11468  405483135  Zhanibek Alimkhanuly         KAZ    male    4/1/93    1.82   
11513  352983859         Zohir Kedache         ALG    male    3/2/86    1.78   
11514  424586991     Zohra Ez Zahraoui         MAR  female  11/18/83    1.65   
11517  495365659    Zoltan Adam Harcsa         HUN    male  11/20/92    1.84   

       weight   sport  gold  silver  bronze  
11465     NaN  boxing     0       0       0  
11468     NaN  boxing     0       0       0  
11513     NaN  boxing     0       0       0  
11514     NaN  boxing     0       0       0  
11517     NaN  boxing     0       0       0

#### canoe (카누)

In [58]:
canoe = op[op.sport == 'canoe']
canoe.tail()

id             name nationality     sex       dob  height  \
11337  282448416           Yue Li         CHN  female  10/27/93    1.70   
11413  978744590  Yusmari Mengana         CUB  female  10/25/93    1.70   
11433  751496609  Yvonne Schuring         AUT  female    1/4/78    1.75   
11452  644379517   Zaza Nadiradze         GEO    male    9/2/93    1.77   
11522  230998458  Zoya Ananchenko         KAZ  female    9/5/96    1.65   

       weight  sport  gold  silver  bronze  
11337    65.0  canoe     0       0       0  
11413    65.0  canoe     0       0       0  
11433    68.0  canoe     0       0       0  
11452    85.0  canoe     0       0       0  
11522    67.0  canoe     0       0       0

#### cycling (싸이클)

In [59]:
cycling = op[op.sport == 'cycling']
cycling.tail()

id            name nationality     sex       dob  height  \
11422  116013385   Yuta Wakimoto         JPN    male   3/21/89    1.81   
11435  503085577    Zac Williams         NZL    male   7/21/95    1.80   
11455  538896901   Zdenek Stybar         CZE    male  12/11/85    1.83   
11471  989437678  Zhao Juan Meng         HKG  female  12/14/89    1.70   
11476  448660080       Zhen Wang         CHN    male   6/26/89    1.72   

       weight    sport  gold  silver  bronze  
11422    74.0  cycling     0       0       0  
11435    89.0  cycling     0       0       0  
11455    72.0  cycling     0       0       0  
11471    60.0  cycling     0       0       0  
11476    65.0  cycling     0       0       0

## 국가의 재정상태, 삶의 수준에 따라 금메달 수가 달라지지 않을까?